scaaml/capture/aes/aes_capture_contexts.py

```diff
@@ -258,6 +258,24 @@ def _capture(scope_class, capture_info: Dict[str, Any], chip_id: int,
         generators.
       dataset (scaaml.io.Dataset): The dataset to save examples to.
     """
+    # Capture using LeCroy.
+    if scope_class == LeCroy:
+        with LeCroy(**capture_info) as lecroy:
```

Comment:
With 3 different scopes, I wonder if code wouldn't look better by rewriting it the following way:
```python
with scope_class(**capture_info) as scope_object:
  scope_object.init(dataset=dataset)
  assert scope_object.scope
  with DefaultCWScope(capture_info.get("cw_scope_serial_number)) as default_cwscope:
    _control_communication_and_capture(
        ....
    )
```

And for each scope class you add the method:
```python
def init(self, **kwargs: Dict[Any, Any]) -> None:
  pass
```
Which for the Lecroy would then be:
```python
def init(self, dataset: scaaml.io.Dataset, **kwargs: Dict[Any, Any]) -> None:
  dataset.capture_info.update(self.scope.get_identity_info())
  dataset.write_config()